### DISCUSSION TITLE
p-dropdown keydown event handling

### DISCUSSION THREAD

**Discussion by umyanl** [primefaces/community#2637] [PrimeNG]
keydown event is handled by procedure onTabKey(event, pressedInInputText = false) in p-dropdown component. Recently I notice event.stopPropagation is added in Fixed #14953 - Add missing stopPropagation. Somehow I'd like to use a dropdown in an editableColumn, this added line prevent tab navigation within table. Is there any solution to go around this?

**Comment by qwadrox** [comment]
@umyanl  
If you could create a stackblitz example showcasing your problem, it would be awesome. :)

**Comment by umyanl** [comment]
https://stackblitz.com/edit/l9bnmz?file=src%2Fapp%2Ftable-row-edit-demo.html

In this example, tap won't be handled at dropdown for Inventory status.

**Reply by qwadrox** [reply]
@umyanl  "this added line prevent tab navigation within table" 

Tab navigation is not perfect but using right and left arrow keys and tab I was able to navigate and edit all fields in the table, including the mentioned dropdown field.

I don't know, is this violating WCAG rules. I found this post: [Keyboard Interaction For Data Grids](https://www.w3.org/WAI/ARIA/apg/patterns/grid/)

>  Once a composite contains focus, keys other than Tab and Shift + Tab enable the user to move focus among its focusable elements. Authors are free to choose which keys move focus inside of a composite

**Meaning arrow keys should be fine, and tab should not be used inside the table since table is a composite widget.** 

https://github.com/user-attachments/assets/8919cc37-38eb-4ef9-9142-6900f824347b
